pkg/agent/util/net_windows.go

```diff
@@ -524,5 +508,76 @@ func HostInterfaceExists(ifaceName string) bool {
 func SetInterfaceMTU(ifaceName string, mtu int) error {
 	cmd := fmt.Sprintf("Set-NetIPInterface -IncludeAllCompartments -InterfaceAlias \"%s\" -NlMtuBytes %d",
 		ifaceName, mtu)
-	return InvokePSCommand(cmd)
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func NewNetRoute(linkIndex int, dstSubnet *net.IPNet, gwAddr net.IP, metric int) error {
+	cmd := fmt.Sprintf("New-NetRoute -InterfaceIndex %v -DestinationPrefix %v -NextHop %v -RouteMetric %d -Verbose",
+		linkIndex, dstSubnet.String(), gwAddr.String(), metric)
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func RemoveNetRoute(linkIndex int, dstSubnet *net.IPNet, gwAddr net.IP) error {
+	cmd := fmt.Sprintf("Remove-NetRoute -InterfaceIndex %v -DestinationPrefix %v -NextHop  %v -Verbose -Confirm:$false",
+		linkIndex, dstSubnet.String(), gwAddr.String())
+	_, err := ps.RunCommand(cmd)
+	return err
+}
+
+func GetNetRoutes(linkIndex int, dstSubnet *net.IPNet) ([]Route, error) {
+	cmd := fmt.Sprintf("Get-NetRoute -InterfaceIndex %d -DestinationPrefix %s -erroraction Ignore | Format-Table -HideTableHeaders",
+		linkIndex, dstSubnet.String())
+	return getNetRoutes(cmd)
+}
```

Comment:
it's weird that you use the `Route` struct for reading the routes but not for writing them. I think `NewNetRoute` and `RemoveNetRoute` should use a parameter of type `Route`

Reply:
@antoninbas gonetsh doesn't use Route as the parameter but I don't have a strong preference.
The only concern is that maybe using Route, developers will have no idea what actual parameters are needed? If you think it is not a problem, I will update the code.
https://github.com/rakelkar/gonetsh/blob/master/netroute/netroute.go#L27

Reply:
I think https://github.com/vishvananda/netlink/blob/master/route_linux.go is a better reference than gonetsh

Reply:
Thank you. Indeed this is better. Updated.